docs/specs/servers/notify/notify-server-api.md

```diff
@@ -2,183 +2,244 @@
+All endpoints expect an `Authorization` header in the form `Authorization: Bearer <notify_api_secret>` using the notify_api_secret associated with a project ID. The secret used should be the one that was generated automatically when configuring notify - with the name `notify_api_secret`.
 
-## Notify
+## Send Notifications
 
-Send a notification to a set of accounts.
+Send 1 or more notifications to accounts. Notifications handled itempotently are deduplicated by `notificationId`.
 
-`POST /notify`
-
-Body:
+`POST /v1/<project-id>/notify`
 
 ```typescript
-{
-  "notification": {
-    "title": string,
-    "body": string,
-    "icon": string,
-    "url": string,
-    "type": string,
+type Body = [{
+  // Idempotency key and ID to track status
+  notificationId?: string | null,
+  notification: {
+    // The notification type ID found in [Notify Config](../../clients/notify/notify-config.md)
+    type: string,
+    title: string,
+    body: string,
+    icon?: string | null,
+    url?: string | null,
   },
-  "accounts": Account[],
-}
+  // The accounts to send this notification to
+  accounts: AccountId[],
+}]
 ```
 
-Account:
+```typescript
+// CAIP-10 account ID
+type AccountId = string
+```
 
 ```typescript
-string
+type Response = {
+  // IDs for the sent notifications returned in the same order as the body. If a `notificationId` wasn't specified in the request, one will be generated.
+  notifications: string[],
+}
 ```
 
-Response: 
+## Notification Status
+
+Get the curent status of a sent notification.
+
+`GET /v1/<project-id>/notification/<notification-id>`
 
 ```typescript
-{
-  "sent": Account[], // notifications sent to subscribers
-  "failed": Failed[], // notifications not sent because there was a failure in delivering
-  "notFound": Account[], // notifications not sent becuase those accounts were not subscribers
+type Response = {
+  status: {
+    [AccountId]: Status,
+  },
 }
 ```
 
-Failed:
-
 ```typescript
-{
-  "account": Account,
-  "reason": string,
+// The status of the notification. Client must be forwards-compatible with new statuses.
+type Status = 
+  // Notifications currently queued to be published
+  "queued" |
+  // Notifications successfully published to subscribers
+  "published" |
+  // Notifications not published becuase those accounts were not subscribers
+  "not-subscribed" |
+  // Notifications not published becuase while those accounts were subscribers, they were not subscribed to the sent notification type
+  "wrong-scope" |
+  // Notifications not published because the account's rate limit was hit
+  "rate-limited"
 }
 ```
 
 ## Subscribers 
 
-Get the list of all accounts currently subscribed to this dapp.
+Get the list of all accounts currently subscribed to this app.
 
-`GET /subscribers`
-
-Response:
+`GET /v1/<project-id>/subscribers`
 
 ```typescript
-Account[]
-``` 
-
-## Subscribe Topic
+type Response = [{
+  account: AccountId,
+  // Notification types subscribed to
+  scope: string[],
+}]
```

Comment:
Make changing the response body easier in the future w/o introducing a new endpoint version:

```suggestion
type Response = {
  subscribers: {
    account: AccountId,
    // Notification types subscribed to
    scope: string[],
  }[],
}
```